### PR TITLE
17338 align dfu result definitions

### PR DESCRIPTION
**Pull request by eivindj-nordic**

### CODE REVIEW

**Review by lemrey — APPROVED** [review]

**Comment by lemrey** [comment]
Actually, something looks off in the diagrams. The mode is wrong (they have execute permissions), and there is a missing end-state for the rollback in case of success?

**Comment by eivindj-nordic** [comment]
> Actually, something looks off in the diagrams. The mode is wrong (they have execute permissions), and there is a missing end-state for the rollback in case of success?

Thanks, updated :)

**Review by lemrey — COMMENTED** [review]
Are we missing
MODEM_DFU_RESULT_VOLTAGE_LOW

**Comment by eivindj-nordic** [comment]
> Are we missing MODEM_DFU_RESULT_VOLTAGE_LOW

Thanks, updated.

**Review by lemrey — APPROVED** [review]

File: nrf_modem/doc/delta_dfu.rst [hunk 1]
```diff
@@ -193,12 +193,12 @@ Checking the result of the update
...
-* ``MODEM_DFU_RESULT_VOLTAGE_LOW`` - The modem did not have sufficient voltage to apply the firmware update. The operation will be retried the next time the modem is started.
+* ``NRF_MODEM_DFU_RESULT_OK`` - The update is successful. The modem will run the updated firmware the next time it boots.
+* ``NRF_MODEM_DFU_RESULT_AUTH_ERROR`` - The update did not take place. The modem will run the original firmware the next time it boots.
+* ``NRF_MODEM_DFU_RESULT_UUID_ERROR`` - The update did not take place. The modem will run the original firmware the next time it boots.
+* ``NRF_MODEM_DFU_RESULT_INTERNAL_ERROR`` - The modem encountered an internal error while updating, and it will not boot to prevent executing unintended operations. For subsequent programming, the modem can only be programmed through the :ref:`nrf_modem_full_dfu_api`.
+* ``NRF_MODEM_DFU_RESULT_HARDWARE_ERROR`` - The modem encountered a hardware error while updating, and it will not boot to prevent executing unintended operations. For subsequent programming, the modem can only be programmed through the :ref:`nrf_modem_full_dfu_api`.
```
Comment (lemrey) [inline nrf_modem/doc/delta_dfu.rst:200]: I had reworded this recently to avoid using `subsequent` and just say `next`

Comment (eivindj-nordic) [reply nrf_modem/doc/delta_dfu.rst]: Thanks, updated.

**Pull request merged**